gtsam/navigation/tests/testCombinedImuFactor.cpp

```diff
@@ -44,207 +44,272 @@ static boost::shared_ptr<PreintegratedCombinedMeasurements::Params> Params() {
+// TEST(CombinedImuFactor, PredictRotation) {
+//   const Bias bias(Vector3(0, 0, 0), Vector3(0, 0, 0)); // Biases (acc, rot)
+//   auto p = testing::Params();
+//   PreintegratedCombinedMeasurements pim(p, bias);
+//   const Vector3 measuredAcc = - kGravityAlongNavZDown;
+//   const Vector3 measuredOmega(0, 0, M_PI / 10.0);
+//   const double deltaT = 0.01;
+//   const double tol = 1e-4;
+//   for (int i = 0; i < 100; ++i)
+//     pim.integrateMeasurement(measuredAcc, measuredOmega, deltaT);
+//   const CombinedImuFactor Combinedfactor(X(1), V(1), X(2), V(2), B(1), B(2), pim);
+
+//   // Predict
+//   const Pose3 x(Rot3::Ypr(0, 0, 0), Point3(0, 0, 0)), x2;
+//   const Vector3 v(0, 0, 0), v2;
+//   const NavState actual = pim.predict(NavState(x, v), bias);
+//   const Pose3 expectedPose(Rot3::Ypr(M_PI / 10, 0, 0), Point3(0, 0, 0));
+//   EXPECT(assert_equal(expectedPose, actual.pose(), tol));
+// }
+
+// /* ************************************************************************* */
+// // Testing covariance to check if all the jacobians are accounted for.
+// TEST(CombinedImuFactor, CheckCovariance) {
+//   auto params = PreintegrationCombinedParams::MakeSharedU(9.81);
+
+//   params->setAccelerometerCovariance(pow(0.01, 2) * I_3x3);
+//   params->setGyroscopeCovariance(pow(1.75e-4, 2) * I_3x3);
+//   params->setIntegrationCovariance(pow(0.0, 2) * I_3x3);
+//   params->setOmegaCoriolis(Vector3::Zero());
+
+//   imuBias::ConstantBias currentBias;
+
+//   PreintegratedCombinedMeasurements actual(params, currentBias);
+
+//   // Measurements
+//   Vector3 measuredAcc(0.1577, -0.8251, 9.6111);
+//   Vector3 measuredOmega(-0.0210, 0.0311, 0.0145);
+//   double deltaT = 0.01;
+
+//   actual.integrateMeasurement(measuredAcc, measuredOmega, deltaT);
+
+//   Eigen::Matrix<double, 15, 15> expected;
+//   expected << 0.01, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,  //
+//       0, 0.01, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,          //
+//       0, 0, 0.01, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,          //
+//       0, 0, 0, 2.50025e-07, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,   //
+//       0, 0, 0, 0, 2.50025e-07, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,   //
+//       0, 0, 0, 0, 0, 2.50025e-07, 0, 0, 0, 0, 0, 0, 0, 0, 0,   //
+//       0, 0, 0, 0, 0, 0, 0.010001, 0, 0, 0, 0, 0, 0, 0, 0,      //
+//       0, 0, 0, 0, 0, 0, 0, 0.010001, 0, 0, 0, 0, 0, 0, 0,      //
+//       0, 0, 0, 0, 0, 0, 0, 0, 0.010001, 0, 0, 0, 0, 0, 0,      //
+//       0, 0, 0, 0, 0, 0, 0, 0, 0, 0.01, 0, 0, 0, 0, 0,          //
+//       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.01, 0, 0, 0, 0,          //
+//       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.01, 0, 0, 0,          //
+//       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.01, 0, 0,          //
+//       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.01, 0,          //
+//       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0.01;
+
+//   // regression
+//   EXPECT(assert_equal(expected, actual.preintMeasCov()));
+// }
```

Comment:
I commented out all the previous tests to make debugging this faster and easier.